### PR TITLE
Keyboard zoom and drawing tool selection

### PR DESCRIPTION
**Pull request by jespino**
Ready to be reviewed

### CODE REVIEW

File: src/uxbox/ui/keyboard.cljs [hunk 1]
```diff
@@ -12,40 +13,27 @@
...
-;; (defn dispatch-key [e]
-;;   (let [event (get event-keys (.-identifier e))]
-;;     (pubsub/publish! event)))
+;; Mixins
 
-;; (defn start-keyboard! []
-;;   (let [handler (KeyboardShortcutHandler. js/document)]
-;;     (doseq [[shortcut key] event-keys]
-;;       (.registerShortcut handler shortcut shortcut))
+(defn ^{:doc "A mixin for capture keyboard events."}
```
Comment (ghost) [inline src/uxbox/ui/keyboard.cljs]: You can put the docstring after between the fn name and argument vector, no need to use metadata directly for `defn`.

**Comment by jespino** [comment]
@dialelo can you review this?

File: src/uxbox/ui/workspace/streams.cljs [hunk 2]
```diff
@@ -18,25 +28,83 @@
...
   (s/to-event-stream selected-tool-bus))
 
+(def keyboard-toggle-grid-stream
+  (s/filter #(= % :toggle-grid) (s/map first workspace-keyboard-bus)))
+
+(def active-grid-stream
+  (s/map #(not @grid?) active-grid-bus))
+
+(def keyboard-zoom-stream
+  (s/filter #(% #{:zoom-in :zoom-out :zoom-reset}) (s/map first workspace-keyboard-bus)))
```
Comment (ghost) [inline src/uxbox/ui/workspace/streams.cljs]: Here we can just use the set as predicate: `(s/filter #{:zoom-in :zoom-out :zoom-reset} ;; ...`

Comment (ghost) [inline src/uxbox/ui/workspace/streams.cljs]: Also, we may want to make these lines more compact for readability:

``` clojure
(s/map #{}
       (s/filter pred bla)
```

**Comment by ghost** [comment]
Besides from the nitpicks looks great to me, i'll merge as soon as we fix those.

File: src/uxbox/ui/workspace/streams.cljs [hunk 3]
```diff
@@ -18,25 +28,86 @@
...
 
+(def keyboard-toggle-grid-stream
+  (s/filter #{:toggle-grid}
+            (s/map first workspace-keyboard-bus)))
+
+(def active-grid-stream
+  (s/map #(not @grid?) active-grid-bus))
+
+(def keyboard-zoom-stream
+  (s/filter #{:zoom-in :zoom-out :zoom-reset}
```
Comment (jespino) [inline src/uxbox/ui/workspace/streams.cljs:80]: This doesn't work, I'm dont understand why, but not filter correctly the events, you can uncomment the lines at the end of the file to see it.

**Comment by ghost** [comment]
Excellent work!

**Pull request merged**